docs/limitrange.md

```diff
@@ -0,0 +1,104 @@
+- **Limits** on the other hand, are a hard stop. A container going over the limit, will be killed.
+
+Resource types for both are:
+-   CPU
+-   Memory
+-   Ephemeral storage
+
+The next question is : how pods with resource and limits are run/scheduled ?
+The scheduler *computes* the amount of CPA and memory requests (using **Requests**) and tries to find a node to schedule it.
+
+Requests and limits can be applied to both Containers and Init Containers.
+- For init containers, the max of each type is taken
+- For containers, it sums all requests/limits for each containers
+
+This means, if you got the following:
+-   initContainer1 : 1 CPU, 100m memory
+-   initContainer2 : 2 CPU, 200m memory
+-   container1 : 1 CPU, 50m memory
+-   container2 : 2 CPU, 250m memory
+-   container3 : 3 CPU, 500m memory
+
+The computation will be:
+-   CPU : 2 (max init containers) + 6 (sum of containers) = 8 CPU
+-   Memory: 200m (max init containers) + 800m (sum of containers) = 1000m (1G)
+
+## Tekton support
+
+The way Limits and Requests works in Kubernetes is because it is assumed that all containers run in parallel, and init container run before, each one after the others.
+
+That assumption — containers running in parallel — is not true in Tekton. They do all start together (because there is no way around this) **but** the *entrypoint hack* is making sure they actually run in sequence and thus there is always only one container that is actually consuming some resource at the same time.
+
+This means, we need to handle limits, request and LimitRanges in a *non-standard* way. Let's try to define that. Tekton needs to take into account all the aspect of the LimitRange : the min/max as well as the default. If there is no default, but there is min/max, Tekton need then to **set** a default value that is between the min/max. If we set the value too low, the Pod won't be able to be created, similar if we set the value too high. **But** those values are set on **containers**, so we **have to** do our own computation to know what request to put on each containers.
+
+
+## A LimitRange is in the namespace
+
+We need to get the default (limits), default requests, min and max values (if they are here).
+One thing to note is that, in the case of a LimitRange being present, we need to **not rely** on the pod mutation webhook that takes the default into account ; what this means is, we need to specify all request and limits ourselves so that the mutation webhook doesn't have any work to do.
+
+- **No default value:** if there is no default value, we need to treat the min as the
+  default. I think that's also what k8s does, at least in our computation.
+- **Default value:** we need to "try" to respect that as much as possible.
+  - `defaultLimit` but no `defaultRequest`, then we set `defaultRequest` to be same as `defaultLimit`.
+  - `defaultRequest` but no `defaultlimit`, then we use the `min` limit as the `defaultLimit`
```

Comment:
curious why we wouldn't use the `max` as the `defaultLimit` ?

Reply:
Wait, I think that's a typo / wrong thing here, I'll fix.

Reply:
Because, in a gist, Requests are used for scheduling (so we need the "minimum") and Limits are used for "killing" the running process (so we kinda need the "maximum")